mithril-client-wasm/src/client_wasm.rs

```diff
@@ -0,0 +1,211 @@
+impl MithrilClient {
+    /// Constructor for wasm client
+    #[wasm_bindgen(constructor)]
+    pub async fn new(aggregator_endpoint: &str, genesis_verification_key: &str) -> MithrilClient {
+        let feedback_receiver = Arc::new(JSBroadcastChannelFeedbackReceiver::new("mithril-client"));
+        let client = ClientBuilder::aggregator(aggregator_endpoint, genesis_verification_key)
+            .add_feedback_receiver(feedback_receiver)
+            .build()
+            .map_err(|err| format!("{err:?}"))
+            .unwrap();
+        MithrilClient { client }
+    }
+
+    /// Call the client to get a snapshot from a digest
+    // #[wasm_bindgen(js_name = get_snapshot)]
+    pub async fn get_snapshot(&self, digest: &str) -> WasmResult {
+        let result = self
+            .client
+            .snapshot()
+            .get(digest)
+            .await
+            .map_err(|err| format!("{err:?}"))?;
+
+        Ok(serde_wasm_bindgen::to_value(&result)?)
+    }
+
+    /// Call the client to get the list of available snapshots
+    // #[wasm_bindgen(js_name = list_snapshots)]
+    pub async fn list_snapshots(&self) -> WasmResult {
+        let result = self
+            .client
+            .snapshot()
+            .list()
+            .await
+            .map_err(|err| format!("{err:?}"))?;
+
+        Ok(serde_wasm_bindgen::to_value(&result)?)
+    }
+
+    /// Call the client to get a mithril stake distribution from a hash
+    // #[wasm_bindgen(js_name = get_mithril_stake_distribution)]
+    pub async fn get_mithril_stake_distribution(&self, hash: &str) -> WasmResult {
+        let result = self
+            .client
+            .mithril_stake_distribution()
+            .get(hash)
+            .await
+            .map_err(|err| format!("{err:?}"))?;
+
+        Ok(serde_wasm_bindgen::to_value(&result)?)
+    }
+
+    /// Call the client for the list of available mithril stake distributions
+    // #[wasm_bindgen(js_name = list_mithril_stake_distributions)]
+    pub async fn list_mithril_stake_distributions(&self) -> WasmResult {
+        let result = self
+            .client
+            .mithril_stake_distribution()
+            .list()
+            .await
+            .map_err(|err| format!("{err:?}"))?;
+
+        Ok(serde_wasm_bindgen::to_value(&result)?)
+    }
+
+    /// Call the client to compute a mithril stake distribution message
+    // #[wasm_bindgen(js_name = compute_mithril_stake_distribution_message)]
+    pub async fn compute_mithril_stake_distribution_message(
+        &self,
+        stake_distribution: JsValue,
+    ) -> WasmResult {
+        let stake_distribution =
+            serde_wasm_bindgen::from_value(stake_distribution).map_err(|err| format!("{err:?}"))?;
+        let result = MessageBuilder::new()
+            .compute_mithril_stake_distribution_message(&stake_distribution)
+            .map_err(|err| format!("{err:?}"))?;
+
+        Ok(serde_wasm_bindgen::to_value(&result)?)
+    }
+
+    ///
```

Comment:
We need to add a comment, just a suggestion, you'll probably have a better idea 😅.
```suggestion
    /// Call the client to validate the computed message
```

Reply:
Done :heavy_check_mark: